relay-filter/src/interface.rs

```diff
@@ -108,3 +110,45 @@ impl Filterable for Replay {
         self.user_agent()
     }
 }
+
+impl Filterable for Span {
+    fn csp(&self) -> Option<&Csp> {
+        // Only for events.
+        None
+    }
+
+    fn exceptions(&self) -> Option<&Values<Exception>> {
+        // Only for events.
+        None
+    }
+
+    fn ip_addr(&self) -> Option<&str> {
+        self.data.value()?.client_address.as_str()
+    }
+
+    fn logentry(&self) -> Option<&LogEntry> {
+        // Only for events.
+        None
+    }
+
+    fn release(&self) -> Option<&str> {
+        self.data.value()?.release.as_str()
+    }
+
+    fn transaction(&self) -> Option<&str> {
+        if self.is_segment.value().map_or(false, |&s| s) {
+            return self.data.value()?.segment_name.as_str();
+        }
```

Comment:
I actually think we can simplify this to 
```suggestion
        self.data.value()?.segment_name.as_str()
```

When we drop a transaction event because of this filter, we also drop its child spans. So dropping every span that can be attributes to this segment name is consistent.